packages/api/src/command/medical/patient/consolidated-get.ts

```diff
@@ -67,8 +69,18 @@ export async function startConsolidatedQuery({
   }
 
   const startedAt = new Date();
+  const requestId = uuidv7();
   const progress: QueryProgress = { status: "processing", startedAt };
 
+  analytics({
+    distinctId: patient.cxId,
+    event: EventTypes.consolidatedQuery,
+    properties: {
+      patientId: patient.id,
+      requestId,
+    },
+  });
```

Comment:
We're sending this one here but there's no code to send another one with the duration. I assume this is the case you mentioned that needed to do Posthog SQL to calculate it, right?

Couldn't we have this on our code (send duration and not have SQL) if check the result of `processRequest()` [here](https://github.com/metriport/metriport/pull/1930/files#diff-5d3732ab972684ed41efa4877291f021f772141baddff08a616b2db2eb0181d4L78) and decide whether consolidated query worked or not?

Reply:
We have the duration one on line 168 and 182

Reply:
As for why it exists look at the thread below